### PR TITLE
ISPN-10600 Create empty XSiteStateTransferManager and related classes

### PR DESCRIPTION
**Pull request by pruivo**
I changed my mind. See comment in https://issues.jboss.org/browse/ISPN-10600?focusedCommentId=13785267&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13785267

https://issues.jboss.org/browse/ISPN-10600

### CODE REVIEW

File: core/src/main/java/org/infinispan/xsite/statetransfer/XSiteStateTransferManager.java [hunk 1]
```diff
@@ -97,7 +97,19 @@
...
+      throw new UnsupportedOperationException();
+   }
 
-   XSiteStateConsumer getStateConsumer();
+   /**
+    * @deprecated since 10.0. Get the {@link XSiteStateConsumer} from the {@link org.infinispan.factories.ComponentRegistry}
+    */
+   @Deprecated
+   default XSiteStateConsumer getStateConsumer() {
+      throw new UnsupportedOperationException();
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/xsite/statetransfer/XSiteStateTransferManager.java]: No need to deprecate these methods as they were introduced in `10.0.0.Beta4` by https://issues.jboss.org/browse/ISPN-10322. Just remove them :slightly_smiling_face:

File: core/src/main/java/org/infinispan/factories/ComponentRegistry.java [hunk 2]
```diff
@@ -421,8 +417,12 @@ public final synchronized void registerVersionGenerator(NumericVersionGenerator
       return asyncInterceptorChain;
    }
 
+   /**
+    * @deprecated since 10.0
+    */
+   @Deprecated
    public ComponentRef<BackupSender> getBackupSender() {
-      return backupSender;
+      return basicComponentRegistry.getComponent(BackupSender.class);
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/factories/ComponentRegistry.java]: Also introduced in Beta4 so can be removed.

File: core/src/main/java/org/infinispan/factories/ComponentRegistry.java [hunk 3]
```diff
@@ -494,8 +494,12 @@ public ByteString getCacheByteString() {
       return rpcManager;
    }
 
+   /**
+    * @deprecated since 10.0
+    */
+   @Deprecated
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/factories/ComponentRegistry.java]: Also introduced in Beta4 so can be removed.

**Comment by ryanemerson** [comment]
@pruivo The reason I introduced the `getXSiteStateTransferManager()`, `getStateProvider()` methods was so that the components would be cached and we would avoid a hash lookup per Command that requires them.

I don't see why you need to change this? But if you do, the deprecations aren't necessary

**Comment by pruivo** [comment]
@ryanemerson it makes sense. I'm going back to my original idea of creating no-op implementations and keep the cache for the remote commands.

**Comment by pruivo** [comment]
@ryanemerson updated!

**Comment by ryanemerson** [comment]
Thanks @pruivo

**Pull request merged**